sdk/java-sdk/src/main/scala/kalix/javasdk/impl/eventsourcedentity/EventSourcedEntityEffectImpl.scala

```diff
@@ -76,6 +76,17 @@ class EventSourcedEntityEffectImpl[S] extends Builder[S] with OnSuccessBuilder[S
     this
   }
 
+  override def deleteEntity(finalEvent: Any): EventSourcedEntityEffectImpl[S] = {
+    if (finalEvent.isInstanceOf[Iterable[_]] || finalEvent.isInstanceOf[java.lang.Iterable[_]]) {
+      throw new IllegalStateException(
+        s"You are trying to emit collection (${finalEvent.getClass}) of events. " +
+        s"Only one final event is supported when deleting the entity.")
```

Comment:
I can imagine a use case where you would like to emit some events and a terminal one together. So sth like:
```
effect
  .emitEvents(E1, E2)
  .deleteEntity(TerminalEvent)
```
but this looks bad. Since in the effect, there is a boolean flag, how about sth like this:
```
effect
  .emitEvents(E1, E2, TerminalEvent)
  .deleteEntity()
```